### ISSUE TITLE
Getting bone roll

### ISSUE THREAD

**Issue by dj-kefir-siorbacz** [Wasserwecken/bvhio#10]
Hi @Wasserwecken ! 

Any idea if `bvhio` could get me a bone roll (i.e. rotation along longitudinal axis)?
I know that Blender arbitrarily assigns the Y axis as the roll axis. How is it in bvhio?

In the worst case I would need to get the bone roll from `joint.Rotation`, right?

**Comment by Wasserwecken** [comment]
Short answer: Sorry nope.

The Bone-Roll ist not arbitrarily set by Blender, you will get a roll with bvhio too, because of two things:
- The restpose-rotations are defined by the position of the child bone.
- Therefore blender and bvhio is restoring the rotation with the Gram-Schmidt process

You could calculate the roll by restoring the rotation by position and comparing it to the actual one

**Comment by dj-kefir-siorbacz** [comment]
Actually, I don't mean the bone-roll as in Blender's edit-mode, but rather simple rotation along the longitudinal axis in pose-mode  (which in blender is bone's Y axis). I don't want to change the rest pose.

**Comment by dj-kefir-siorbacz** [comment]
If `joint.Rotation` stores joint's local rotation, then it should be as easy as converting to Euler and getting the value of a single axis, right?

**Comment by Wasserwecken** [comment]
Im not sure, you may want to evaluate that on your own. Be aware than euler values depend on order being applied.
If you want to convert the quaternions of `joint.Rotation` to a specific set of euler angles you may want to have a look at here:
https://github.com/Wasserwecken/spatial-transform#euler-angles-conversions

**Comment by dj-kefir-siorbacz** [comment]
I think I will also try with `joint.Up` and `joint.RestPose.Up` (or `Right` or `Forward`). Thanks for help!

**Comment by dj-kefir-siorbacz** [comment]
Hi @Wasserwecken ! 
Sorry for torturing you with this topic. I'm coming back as I acquired a higher understanding of what's going on (imho :D).

(Just a reminder: this whole issue is about getting the pose/keyframe rotation of a bone, not rest pose rotation)
(I saw people calling this longitudinal rotation a "twist" rather than "roll", so let's switch to that.)

I guess we can get this twist using twist-swing decomposition of a quaternion (https://allenchou.net/2018/05/game-math-swing-twist-interpolation-sterp/).

I also tested empirically (but I can't say it has been an exhaustive test nor do I have any formal mathetatical proof for that, yet) that if you convert your euler rotation order to YXZ or YZX (at least that's how they are called in Blender), then the twist angle can be easily extracted as the Euler's `y` component (i.e. the orientation of the joint/object with respect to the world/global axes does not influence its local Y rotation, which is not the case in ZYX, ZXY, XYZ, XZY rotation orders). Ofc, I'm assuming that the longitudinal axis is the Y axis. 
Therefore you might not need to do the swing-twist decomposition of the quaternion representation.
